pkg/apis/cluster/v1alpha1/cluster_types.go

```diff
@@ -236,9 +234,9 @@ type ClusterStatus struct {
 	// Format should be: `https://host:port`
 	// +optional
 	APIEndpoint string `json:"apiEndpoint,omitempty"`
-	// Kubeconfig represents the secret that contains the credential to access this cluster.
+	// KubeconfigSecretRef represents the secret that contains the credential to access this cluster.
 	// +optional
-	KubeConfig *infrav1alpha1.SecretReference `json:"kubeconfig,omitempty"`
+	KubeconfigSecretRef *corev1.SecretReference `json:"kubeconfigSecretRef,omitempty"`
```

Comment:
useless property named `Namespace` on `corev1.SecretReference`?

Reply:
Yes, there is a implicit meaning if unset, it is within the same namespace as the cluster

Reply:
I think always keep in same namespace for security?